src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -213,10 +214,19 @@ public boolean isUpdateSuccessful() {
             .collect(Collectors.toList());
 
         // Create piece records for item id's which do not have piece records yet
-        itemIds.stream()
-               .filter(id -> !existingItemIds.contains(id))
-               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
-
+        if(itemIds!=null) {
+	        itemIds.stream()
+	               .filter(id -> !existingItemIds.contains(id))
+	               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
+        }
+        
+        // Calculate total quantity and create pieces when item record does not exists
```

Comment:
I see 2 problems here:
1) For a case where we are creating pieces corresponding to itemIds, the below block of code will be executed after already creating the pieces in line 220, so essentially creating twice the pieces. should this be in else block?
2) If we are creating pieces irrespective of inventory interaction, Why do we need to subtract  calculateInventoryItemsQuantity(compPOL) from the total quantity? The total quantity should just be physical+electronic

Reply:
Thanks for catching these @KVupp. I have updated code to handle them.